packages/engine/lib/orchestrator/src/experiment.rs

```diff
@@ -95,6 +95,23 @@ pub struct ExperimentConfig {
     clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
     )]
     pub num_workers: usize,
+
+    /// Allows a JavaScript runner to grow to some initial size in megabytes before triggering
+    /// garbage collections.
+    ///
+    /// This is useful when it is known that experiments need a certain minimum heap to run to
+    /// avoid repeatedly invoking the garbage collector when growing the heap.
+    /// When used in the wrong conditions this could waste memory.
```

Comment:
Do we any more about this? If we don't then no worries, I'm not sure we have to teach V8 to users, but I want to know if we understand what this number actually does. Is it just for the old gen space? Is it for the total heap? Does this continue to affect garbage collections later on or is it just the threshold where the **first** collection happens?

If we find a stack-overflow or forum comment or something that explains this that'd be great. But not trying to get us to spend too much more time investigating.

Reply:
Can we also be explicit that providing `None` uses V8 defaults (**not** asking us to say what the defaults are, just that they're used)

Reply:
> Do we any more about this?
I don't understand.

I linked a stack-overflow comment in a few comments, ([link](https://stackoverflow.com/a/58041256)). It explains what this does and what could potentially be a situation where it's valuable.

Reply:
Well first of all I'd highlight that a stack-overflow link in some resolved comment chain on a PR isn't exactly a source of knowledge we'll easily find again. Lets make an effort to persist these in a useful place.

Secondly I tried to be clear with some specific guiding questions:
>  Is it just for the old gen space? Is it for the total heap?

As an example, your link seems to suggest the argument is actually --initial-_**old-space**_-memory, it's not just about the heap.

The link also gives some examples of what the right and wrong conditions are, the comment and description we've written do not convey any of that. For instance, it specifically gives a warning that setting the value generally isn't a great idea and it's not on by default. We haven't said that, we've just got some generic not-useful warnings about "When used in the wrong conditions this could waste memory"

Reply:
There might be a misunderstanding, I did understand most of your comment, just not the first question which might be missing a word? Maybe not?
So I wanted clarification on that first before addressing the rest.

Reply:
Rephrased in https://github.com/hashintel/hash/pull/446/commits/05d3b82d61885f592f140324d30817805fafa106